ocaml/tests/test_sr_allowed_operations.ml

```diff
@@ -0,0 +1,79 @@
+(*
+ * Copyright (C) 2006-2012 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Test_common
+
+(* Helpers for testing Xapi_sr_operations.assert_operation_valid *)
+
+let setup_test ~__context ?sm_fun ?vdi_fun () =
+  let run f x = match f with Some f -> ignore(f x) | None -> () in
+  let _sm_ref = make_sm ~__context () in
+  let sr_ref = make_sr ~__context () in
+  let (_: API.ref_PBD) = make_pbd ~__context ~sR:sr_ref () in
+  run sm_fun _sm_ref;
+  let sr_record = Db.SR.get_record ~__context ~self:sr_ref in
+  sr_ref, sr_record
+
+let check_same_error_code =
+  let open Alcotest in
+  let open Alcotest_comparators in
+  check (option error_code) "Same error code"
```

Comment:
You can just use `Alcotest.check_raises` for this, e.g.

```
Alcotest.check_raises
  "check blocked during RPU"
  Api_errors.(Server_error (not_supported_during_upgrade, []))
  (fun () -> Xapi_sr_operations.assert_operation_valid ~__context
                   ~self:sr_ref ~op:`vdi_enable_cbt)
```

Reply:
true, but I'm following the same pattern as in `test_vdi_allowed_operations.ml`, and it'll make it easier for further tests of the allowed_operations errors to ignore the params of the exception.